packages/f-services/src/validations.js

```diff
@@ -1,19 +1,31 @@
+/**
+ * Returns a deep object by traversing following the provided path.
+ *
+ * @param {Object} obj The object to traverse.
+ * @param {Array} path The path used to traverse the object.
+ * @return {Object} The object at the end of the path.
+ */
+const getDeepObjectByPath = (obj, path) => path.reduce((o, key) => (o && o[key] ? o[key] : null), obj);
+
 /**
  * Returns an object containing arrays of the names of valid and invalid validation rules.
  *
  * @param {Object} $v The Vuelidate model
  * @return {Object} Containing two fields: validFields and invalidFields.
  */
 const getFormValidationState = $v => {
-    const fields = $v.$params;
     const invalidFields = [];
     const validFields = [];
+    const flattenedParams = $v.$flattenParams();
+
+    flattenedParams.forEach(param => {
+        const deepObjectProperty = getDeepObjectByPath($v, param.path);
+        const propertyFullPath = param.path.join('.');
 
-    Object.keys(fields).forEach(key => {
-        if ($v[key].$invalid) {
-            invalidFields.push(key);
-        } else {
-            validFields.push(key);
+        if (deepObjectProperty.$invalid && invalidFields.indexOf(propertyFullPath) === -1) {
```

Comment:
It's probably either worth extracting this logic out to another method so it can be commented on or adding a comment here as it's quite hard to scan and see what's being checked here.